### ISSUE TITLE
Make sure everything runs on the main thread

### ISSUE THREAD

**Issue by Dadido3** [sciter-sdk/go-sciter#201]
The lib needs to make sure that main() is running exclusively on the main thread. Otherwise strange behavior, freezes or crashes may happen. (Or on macOS it may just terminate the program)

There is no problem if there are no other goroutines running, but as soon as goroutines are started by some lib or the program, there is no guarantee anymore that window creation or the message pump loop will run on the main thread. (Even if they are called from main())

See [here](https://sciter.com/forums/topic/sciter-crashes-with-multiple-windows/) for some details.

### Possible solution:
Add `runtime.LockOSThread()` to one `init()` function of your lib. Init will always be called from the main thread, and once it's locked to the current thread, `main()` will also be running exclusively on the main thread.

Basically similar to [this](https://github.com/golang/go/wiki/LockOSThread), but without the channel to run functions on main.

**Comment by enify** [comment]
Thinks! that help me a lot, it solves my app hang problem on Windows

**Comment by pravic** [comment]
Could someone check how #282 works?
